code/modules/emoji/emoji.dm

```diff
@@ -0,0 +1,29 @@
+/proc/emoji_parse(text)
```

Comment:
I think it'd be a lot better to just use `findtext(text, ":thinking;")` and similar directly

Reply:
I had to think about it for a moment but that would definitely solve some issues with this parsing algo (and be much smaller as a side effect). I'll rewrite it in a bit when I get home to findtext() iterate over a static emoji list.